pkg/controller/nodeconfig/resource.go

```diff
@@ -116,6 +117,157 @@ func makeNodeConfigClusterRoleBinding() *rbacv1.ClusterRoleBinding {
 	}
 }
 
+func makeNodeDiskSetupDaemonSet(nc *scyllav1alpha1.NodeConfig, operatorImage string) *appsv1.DaemonSet {
+	if nc.Spec.LocalDiskSetup == nil || nc.Spec.LocalDiskSetup.Disabled {
+		return nil
+	}
+
+	labels := map[string]string{
+		"app.kubernetes.io/name":   naming.NodeConfigAppName,
+		naming.NodeConfigNameLabel: nc.Name,
+	}
+
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-disk-setup", nc.Name),
+			Namespace: naming.ScyllaOperatorNodeTuningNamespace,
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(nc, nodeConfigControllerGVK),
+			},
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: naming.NodeConfigAppName,
+					NodeSelector:       nc.Spec.Placement.NodeSelector,
+					Affinity:           &nc.Spec.Placement.Affinity,
+					Tolerations:        nc.Spec.Placement.Tolerations,
+					Volumes: []corev1.Volume{
+						{
+							Name: "hostfs",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+									Type: pointer.Ptr(corev1.HostPathDirectory),
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:            naming.NodeConfigAppName,
+							Image:           operatorImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command: []string{
+								"/usr/bin/bash",
+								"-euExo",
+								"pipefail",
+								"-c",
+							},
+							Args: []string{
+								`
+shopt -s inherit_errexit
+
+cd "$( mktemp -d )"
+
+# Mount /usr to use container binaries
+mkdir -p ./usr
+mount --rbind /usr ./usr
+
+# Mount dependencies of container binaries
+for f in bin lib lib64 sbin; do
+	ln -sf "usr/${f}" "./${f}"
```

Comment:
just to be sure we don't depend on the OS too much, can we use mount here as well, possibly merge it into a single loop?

Reply:
no longer there